source/agora/common/BanManager.d

```diff
@@ -210,8 +210,9 @@ public class BanManager
 
     ***************************************************************************/
 
-    private Status* get (Address address)
+    private Status* get (Address address) @trusted nothrow pure
     {
+        scope(failure) assert(0);  // it will never throw
```

Comment:
`require` is not `nothrow` ??

Reply:
It creates an item if it doesn't exist. The naming of `require` is very counter-intuitive.